### PR TITLE
fix: authd will now wait for examplebroker to start

### PR DESCRIPTION
**Pull request by denisonbarbosa**
If built using the `withexamplebroker` tag, authd is expected to configure and run the examplebroker as a "sub" service. We used to delegate the starting process to a go routine, but it created problems in some test environments (as the examplebroker could take too long to start). 

Now starting the examplebroker is a blocking call and the daemon will wait until the broker is up and running.

### CODE REVIEW

File: internal/brokers/withexamples.go [hunk 1]
```diff
@@ -27,23 +26,17 @@ func useExampleBrokers() (string, func(), error) {
...
-		if err = examplebroker.StartBus(ctx, cfgPath); err != nil {
-			log.Errorf(ctx, "Error starting examplebroker: %v", err)
-		}
-	}()
-
-	// Give some time for the broker to start
-	time.Sleep(time.Second)
+	conn, err := examplebroker.StartBus(cfgPath)
+	if err != nil {
+		os.RemoveAll(cfgPath)
```
Comment (didrocks) [inline internal/brokers/withexamples.go:31]: Now that it’s not in a defer call, we should probably add a warning if we can’t remove the directory (here and below).

**Review by didrocks — APPROVED** [review]
Indeed, that makes way more sense! Well done. Just one thing to fix (which was already there) before going on merging it!

**Comment by 3v1n0** [comment]
Looks good, I've pushed this to https://launchpad.net/~3v1n0/+archive/ubuntu/authd/+packages so let's see how it behaves in the situations we had failures...

As per this, can we also maybe drop some other sleeps we've in the integration tests?

File: internal/brokers/withexamples.go [hunk 2]
```diff
@@ -28,14 +28,18 @@ func useExampleBrokers() (string, func(), error) {
 
 	conn, err := examplebroker.StartBus(cfgPath)
 	if err != nil {
-		os.RemoveAll(cfgPath)
+		if err := os.RemoveAll(cfgPath); err != nil {
+			log.Errorf(context.Background(), "Failed to remove the broker configuration directory: %v", err)
```
[the remark this answers is not in the document]
Comment (denisonbarbosa) [reply internal/brokers/withexamples.go:32]: oops, you are correct. I'll update it.

**Review by didrocks — APPROVED** [review]

**Pull request merged**